### PR TITLE
Debugging feature examples update

### LINKED ISSUE #1517: Doc Migration AutoIvc -> Features Debugging
### Summary of Issue

features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.QuadraticComp
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.setUp
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.test_list_inputs
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.test_list_outputs
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.test_list_explicit_outputs
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.test_list_implicit_outputs
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.test_list_return_value
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.test_for_docs_list_no_values
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.test_list_prom_names
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.test_for_docs_list_includes_excludes
features/debugging/listing_variables.rst:    openmdao.test_suite.test_examples.test_betz_limit.TestBetzLimit.test_betz_with_var_tags
features/debugging/listing_variables.rst:    openmdao.core.tests.test_impl_comp.ListFeatureTestCase.test_list_residuals_with_tol
features/debugging/listing_variables.rst:    openmdao.core.tests.test_expl_comp.ExplCompTestCase.test_for_feature_docs_list_vars_options
features/debugging/listing_variables.rst:    openmdao.core.tests.test_expl_comp.ExplCompTestCase.test_for_docs_array_list_vars_options
features/debugging/listing_variables.rst:  openmdao.core.tests.test_distrib_list_vars.MPIFeatureTests.test_distribcomp_list_feature
features/debugging/listing_variables.rst:    openmdao.core.tests.test_problem.TestProblem.test_feature_list_problem_vars
features/debugging/debugging_drivers.rst:      openmdao.drivers.tests.test_scipy_optimizer.TestScipyOptimizeDriverFeatures.test_feature_debug_print_option
features/debugging/debugging_drivers.rst:      openmdao.drivers.tests.test_scipy_optimizer.TestScipyOptimizeDriverFeatures.test_feature_debug_print_option_totals
features/debugging/debugging_solvers.rst:      openmdao.solvers.tests.test_solver_debug_print.TestNonlinearSolvers.test_solver_debug_print_feature

### Issue Type

- [x ] Bug
- [ ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>

### Environment

<For bugs, otherwise N/A>

Operating System: <i.e. OS X 10.14.6, Windows 10, Ubuntu 16.04>

Python environment: <i.e. Anaconda Python 3.7.1>

Packages: <versions of Dymos, OpenMDAO, and other pertinent Python packages. You can just paste the output of `pip freeze` or `conda list` here.>

### PR DESCRIPTION
**Pull request by robfalck**
### Summary

Update the debugging feature docs to utilize the autoIVC capability where possible.

### Related Issues

- Resolves #1517 

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: openmdao/solvers/tests/test_solver_debug_print.py [hunk 1]
```diff
@@ -174,9 +171,11 @@ def test_solver_debug_print_feature(self):
         nl.options['debug_print'] = True
         nl.options['err_on_non_converge'] = True
 
-        # set some poor initial guesses so that we don't converge
-        p['circuit.n1.V'] = 10.
-        p['circuit.n2.V'] = 1e-3
+        # set some poor initial guesses so that we don't converg
```
Comment (Kenneth-T-Moore) [inline openmdao/solvers/tests/test_solver_debug_print.py:174]: converg -> converge

File: openmdao/solvers/tests/test_solver_debug_print.py [hunk 2]
```diff
@@ -159,13 +159,10 @@ def test_solver_debug_print_feature(self):
         p = om.Problem()
         model = p.model
 
-        model.add_subsystem('ground', om.IndepVarComp('V', 0., units='V'))
-        model.add_subsystem('source', om.IndepVarComp('I', 0.1, units='A'))
+        # model.add_subsystem('ground', om.IndepVarComp('V', 0., units='V'))
+        # model.add_subsystem('source', om.IndepVarComp('I', 0.1, units='A'))
```
Comment (swryan) [inline openmdao/solvers/tests/test_solver_debug_print.py:163]: delete comments?

Comment (robfalck) [reply openmdao/solvers/tests/test_solver_debug_print.py]: resolved

File: openmdao/solvers/tests/test_solver_debug_print.py (hunk 1 above)
Comment (robfalck) [reply openmdao/solvers/tests/test_solver_debug_print.py]: resolved

**Pull request merged**